tsdb/head.go

```diff
@@ -2057,55 +2062,80 @@ func (s *memSeries) head() *memChunk {
 	if s.headChunk == nil {
 		return nil
 	}
-	return &s.headChunk.memChunk
+	return &s.headChunk.chunk
 }
 
 type memChunk struct {
 	chunk            chunkenc.Chunk
 	minTime, maxTime int64
 }
 
-type memChunkList struct {
-	memChunk
-	prev *memChunkList // older element, nil for the last (oldest) element on the list
-	next *memChunkList // newer element, nil for the first (most recent) element on the list
+type memChunkList[T any] struct {
```

Comment:
Why is this a generic? Can we keep the `prev` with memChunk itself?

Reply:
That's done and ready for review